lib/iris/_data_manager.py

```diff
@@ -365,6 +386,14 @@ def data(self, data):
         # Always reset the realised dtype, as the managed data has changed.
         self._realised_dtype = None
 
+        # Reset the fill-value appropriately.
+        if ma.isMaskedArray(data):
+            # Align the data manager fill-value with the numpy fill-value.
```

Comment:
I don't think this is a great idea, should probably just clear fill_value when assigning data.
If you have a fill value, you can use `replace` instead.

Actually, why do we need a separate data.setter at all now ?
As this is a totally internal API, not a public, I expect we could now use `replace` for all data assignment ?